dockerscripts/check-user.go

```diff
@@ -0,0 +1,136 @@
+// +build ignore
+
+/*
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/user"
+	"syscall"
+
+	"github.com/minio/cli"
+	minio "github.com/minio/minio/cmd"
+)
+
+var defaultUserGroup string
+
+func init() {
+	username := os.Getenv("MINIO_USERNAME")
+	groupname := os.Getenv("MINIO_GROUPNAME")
+	defaultUserGroup = username + ":" + groupname
+}
+
+func getUserGroup(path string) (string, error) {
+	fi, err := os.Stat(minio.PathJoin(path, ".minio.sys"))
+	if err != nil {
+		// Fresh directory we should default to what was requested by user.
+		if os.IsNotExist(err) {
+			cmd := exec.Command("chown", "-R", defaultUserGroup, path)
+			if err = cmd.Run(); err != nil {
+				return "", err
+			}
+			return defaultUserGroup, nil
+		}
+		return "", err
+	}
+	stat, ok := fi.Sys().(*syscall.Stat_t)
+	if !ok {
+		// Unable to figure out uid/gid, default to defaultUserGroup
+		return defaultUserGroup, nil
+	}
+	u, err := user.LookupId(fmt.Sprintf("%d", stat.Uid))
+	if err != nil {
+		return fmt.Sprintf("%d:%d", stat.Uid, stat.Gid), nil
+	}
+	g, err := user.LookupGroupId(fmt.Sprintf("%d", stat.Gid))
+	if err != nil {
+		return fmt.Sprintf("%d:%d", stat.Uid, stat.Gid), nil
+	}
+	return fmt.Sprintf("%s:%s", u.Username, g.Name), nil
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Flags = append(minio.ServerFlags, minio.GlobalFlags...)
+	app.Action = func(ctx *cli.Context) {
+		// Fetch address option
+		serverAddr := ctx.GlobalString("address")
+		if serverAddr == "" || serverAddr == ":9000" {
+			serverAddr = ctx.String("address")
+		}
+		if ctx.Args().First() == "help" {
+			cli.ShowCommandHelpAndExit(ctx, "check-user", 1)
+		}
+		if ctx.Args().First() != "minio" {
+			cli.ShowCommandHelpAndExit(ctx, "check-user", 1)
+		}
+		args := cli.Args(ctx.Args().Tail())
+		if !args.Present() {
+			cli.ShowCommandHelpAndExit(ctx, "check-user", 1)
+		}
+		var ug string
+		var err error
+		switch args.First() {
+		case "gateway":
+			args = cli.Args(args.Tail())
+			if args.First() != "nas" {
+				fmt.Println(os.Getenv("MINIO_USERNAME"))
+				return
+			}
```

Comment:
Tried S3 gateway, it runs as new user, but prints a message saying otherwise

```
docker run -p 9000:9000 -e MINIO_ACCESS_KEY=Q3AM3UQ867SPQQA43P2F -e MINIO_SECRET_KEY=zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG -e MINIO_USERNAME=user -e MINIO_GROUPNAME=test custom-minio:latest gateway s3 https://play.minio.io:9000 Q3AM3UQ867SPQQA43P2F zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG
Requested username/group hello:now cannot be used
Found existing data with user hello, we will continue and use hello instead.

Endpoint:  http://172.17.0.2:9000  http://127.0.0.1:9000

Browser Access:
   http://172.17.0.2:9000  http://127.0.0.1:9000

Object API (Amazon S3 compatible):
   Go:         https://docs.min.io/docs/golang-client-quickstart-guide
   Java:       https://docs.min.io/docs/java-client-quickstart-guide
   Python:     https://docs.min.io/docs/python-client-quickstart-guide
   JavaScript: https://docs.min.io/docs/javascript-client-quickstart-guide
   .NET:       https://docs.min.io/docs/dotnet-client-quickstart-guide


```

Reply:
Looks like the error message should be correct..